### PR TITLE
Add LoadMore component

### PR DESCRIPTION
**Pull request by josephfusco**
Signed-off-by: Joe Fusco <joe.fusco@wpengine.com>

These changes implement relay style pagination for the following content types

- projects
- posts
- categories
- tags

Categories & tags are nearly identical components so those were consolidated, similar to https://github.com/wp-graphql/wpgraphql.com/blob/2350967adee31fc91190928bc95d370192fc1dd0/src/wp-templates/archive.js

### CODE REVIEW

File: components/LoadMore/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    isLoading={isLoading}
+    fetchMore={fetchMore}
+  />
+);
+```
+
+## Props
+
+- @param {boolean} props.hasNextPage Flag to use if there are more results to load.
+- @param {string} props.endCursor The next pagination cursor string.
```
Comment (blakewilson) [inline components/LoadMore/README.md:20]: `endCursor` is not detailed below

**Review by blakewilson — COMMENTED** [review]
Nice work @josephfusco! I think we'll also need to implement this functionality on the `category` and `tag` templates as well.

File: components/LoadMore/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- @param {string} props.isLoading Flag that indicates whether the pagination is loading.
```
Comment (blakewilson) [inline components/LoadMore/README.md:21]:
```suggestion
- @param {boolean} props.isLoading Flag that indicates whether the pagination is loading.
```

File: wp-templates/archive.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  NavigationMenu,
+  Posts,
+  FeaturedImage,
+  SEO,
+} from '../components';
+import appConfig from 'app.config';
+
+export default function Archive(props) {
+  const { uri, name, __typename } = props.data.nodeByUri;
+  const { data, loading, fetchMore } = useQuery(Archive.query, {
```
Comment (blakewilson) [inline wp-templates/archive.js:21]: 💯

File: wp-templates/index.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
-import category from './category';
-import tag from './tag';
+// import category from './category';
+// import tag from './tag';
```
Comment (blakewilson) [inline wp-templates/index.js:2]:
```suggestion
```

File: pages/projects/index.js [hunk 5]
```diff
@@ -15,25 +16,39 @@ import {
 import { getNextStaticProps } from '@faustwp/core';
 import { pageTitle } from 'utilities';
 import { BlogInfoFragment } from 'fragments/GeneralSettings';
+import appConfig from 'app.config';
 
 export default function Page() {
```
Comment (blakewilson) [inline pages/projects/index.js:21]: Is there a reason why we are using a Next.js file based page instead of the template hierarchy solution here?

Comment (josephfusco) [reply pages/projects/index.js:21]: @blakewilson Not that I'm aware of, more so a remnant of legacy faust

**Review by theodesp — APPROVED** [review]

**Pull request merged**